docs/src/userguide/config/event_config.rst

```diff
@@ -299,10 +299,48 @@ Example configuration for `Google BigQuery`_, by specifying a query in the BigQu
 
     Inherited from :ref:`global event input config <skip-klio-read>`.
 
+
+.. _event-config-avro-read:
+
+Avro
+^^^^
+
+*Mode: Batch*
+
+``KlioReadFromAvro`` reads records in from the provided avro location.
+If the records' schema include an ``element`` field,
+the KlioMessage ``data.element`` will be set to the value.
+If there is no ``element`` field on the records read in from the avro,
+the entire record will be cast to bytes and stuffed into
+the KlioMessage ``data.element`` field.
+
+Example configuration for reading elements from a file:
+
+.. code-block:: yaml
+
+    name: my-cool-batch-job
+    pipeline_options:
+        streaming: False
+    job_config:
+        events:
+            inputs:
+                - type: avro
+
+Example pipeline reading in elements from an avro file:
```

Comment:
The other IOs described on this page do not have examples of using the transform itself. It may be useful to move this into the docstrings of the respective classes themselves.